frontend/src/utils/createShoot.js

```diff
@@ -194,37 +196,61 @@ export function splitCIDR (cidrToSplitStr, numberOfNetworks) {
   return cidrArray
 }
 
-export function getDefaultZonesNetworkConfiguration (zones, infrastructureKind, maxNumberOfZones) {
+export function getDefaultNetworkConfigurationForAllZones (numberOfZones, infrastructureKind, workerCIDR) {
   switch (infrastructureKind) {
     case 'aws': {
-      const zoneNetworksAws = splitCIDR(workerCIDR, maxNumberOfZones)
-      return map(zones, (zone, index) => {
-        const bigNetWorks = splitCIDR(zoneNetworksAws[index], 2)
+      const zoneNetworksAws = splitCIDR(workerCIDR, numberOfZones)
+      return map([...Array(numberOfZones).keys()], zoneIndex => {
+        const bigNetWorks = splitCIDR(zoneNetworksAws[zoneIndex], 2)
         const workerNetwork = bigNetWorks[0]
         const smallNetworks = splitCIDR(bigNetWorks[1], 2)
         const publicNetwork = smallNetworks[0]
         const internalNetwork = smallNetworks[1]
         return {
-          name: zone,
           workers: workerNetwork,
           public: publicNetwork,
           internal: internalNetwork
         }
       })
     }
     case 'alicloud': {
-      const zoneNetworksAli = splitCIDR(workerCIDR, maxNumberOfZones)
-      return map(zones, (zone, index) => {
+      const zoneNetworksAli = splitCIDR(workerCIDR, numberOfZones)
+      return map([...Array(numberOfZones).keys()], zoneIndex => {
         return {
-          name: zone,
-          workers: zoneNetworksAli[index]
+          workers: zoneNetworksAli[zoneIndex]
         }
       })
     }
   }
 }
 
-export function getZonesNetworkConfiguration (oldZonesNetworkConfiguration, newWorkers, infrastructureKind, maxNumberOfZones) {
+export function getDefaultZonesNetworkConfiguration (zones, infrastructureKind, maxNumberOfZones) {
+  const zoneConfigurations = getDefaultNetworkConfigurationForAllZones(maxNumberOfZones, infrastructureKind, defaultWorkerCIDR)
+  if (!zoneConfigurations) {
+    return undefined
+  }
+  return map(zones, (zone, index) => {
+    return {
+      name: zone,
+      ...zoneConfigurations[index]
+    }
+  })
+}
+
+export function findFreeNetworks (existingZonesNetworkConfiguration, workerCIDR, infrastructureKind, maxNumberOfZones) {
+  for (let numberOfZones = maxNumberOfZones; numberOfZones >= existingZonesNetworkConfiguration.length; numberOfZones--) {
+    const newZonesNetworkConfiguration = getDefaultNetworkConfigurationForAllZones(numberOfZones, infrastructureKind, workerCIDR)
+    const freeZoneNetworks = filter(newZonesNetworkConfiguration, networkConfiguration => {
+      return !some(existingZonesNetworkConfiguration, networkConfiguration)
+    })
+    if (newZonesNetworkConfiguration && freeZoneNetworks && newZonesNetworkConfiguration.length - freeZoneNetworks.length === existingZonesNetworkConfiguration.length) {
```

Comment:
freeZoneNetworks.length?
extract `newZonesNetworkConfiguration.length - freeZoneNetworks.length === existingZonesNetworkConfiguration.length` to variable?

Reply:
done